### PR TITLE
Adding data types to tsets and links

### PR DESCRIPTION
**Pull request by nirandaperera**

### CODE REVIEW

**Comment by nirandaperera** [comment]
Added types to Links so far. Will add types to Sets now

**Comment by nirandaperera** [comment]
@supunkamburugamuve I think this is now complete. Could you please review it?

File: twister2/api/src/java/edu/iu/dsc/tws/api/tset/sets/TupleTSet.java [hunk 1]
```diff
@@ -54,4 +55,22 @@
...
+  TupleTSet<K, V> withDataType(MessageType dataType);
+
+  /**
+   * Sets the data type of the {@link TupleTSet} output. This will be used in the packers for efficient
+   * SER-DE operations in the following {@link TLink}s
+   *
+   * @param keyType data type as a {@link MessageType}
+   * @return this {@link TupleTSet}
+   */
+  TupleTSet<K, V> withKeyType(MessageType keyType);
```
Comment (supunkamburugamuve) [inline twister2/api/src/java/edu/iu/dsc/tws/api/tset/sets/TupleTSet.java]: I think we can just have a data type. For key case, we can introduce a MessageType with two types

**Comment by nirandaperera** [comment]
created a new PR for this #850

**Pull request closed without merging**